### PR TITLE
dbt-teradata 1.8.0 changes

### PR DESCRIPTION
**Pull request by tallamohan**
## What are you changing in this pull request and why?
Made changes in the setup and config pages w.r.t the dbt-teradata 1.8.0.

### CODE REVIEW

File: website/docs/docs/core/connect-data-platform/teradata-setup.md [hunk 1]
```diff
@@ -225,6 +230,11 @@ For using cross-DB macros, teradata-utils as a macro namespace will not be used,
 
   `last_day` in `teradata_utils`, unlike the corresponding macro in `dbt_utils`, doesn't support `quarter` datepart.
 
+## dbt - unit tests
+dbt-teradata 1.8.0 and above versions support unit tests, allowing validation of SQL modeling logic on a small set of static inputs before production. This enhancement facilitates test-driven development, improving developer efficiency and code reliability.
+
+More information on dbt unit tests feature can be found [here](https://docs.getdbt.com/docs/build/unit-tests).
+
```
Comment (dataders) [inline website/docs/docs/core/connect-data-platform/teradata-setup.md]: hi @tallamohan -- please consider wrapping this (and any other relevant sections in a `<VersionBlock>`! More info on [versioning blocks of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content)

Comment (tallamohan) [reply website/docs/docs/core/connect-data-platform/teradata-setup.md]: Hi @dataders , I have added VersionBlock like the below:

`<VersionBlock firstVersion="1.8.0">
    dbt-teradata 1.8.0 and above versions support unit tests, allowing validation of SQL modeling logic on a small set of static inputs before production. This enhancement facilitates test-driven development, improving developer efficiency and code reliability.
    More information on dbt unit tests feature can be found [here](https://docs.getdbt.com/docs/build/unit-tests).
</VersionBlock>`

But in the preview, It is showing up like this :
Screenshot 2024-06-25 110402

Can you please help me with this?

Thanks,
Mohan

Comment (mirnawong1) [reply website/docs/docs/core/connect-data-platform/teradata-setup.md]: hey @tallamohan , i've fixed this up for you and it should be rendering correctly now! here's the [staging link](https://docs-getdbt-k3oax5aq5-dbt-labs.vercel.app/docs/core/connect-data-platform/teradata-setup)

Screenshot 2024-06-25 at 09 23 52

File: website/docs/docs/core/connect-data-platform/teradata-setup.md [hunk 2]
```diff
@@ -225,6 +230,12 @@ For using cross-DB macros, teradata-utils as a macro namespace will not be used,
 
   `last_day` in `teradata_utils`, unlike the corresponding macro in `dbt_utils`, doesn't support `quarter` datepart.
 
+<VersionBlock firstVersion="1.8">
+
+dbt-teradata 1.8.0 and above versions support unit tests, allowing validation of SQL modeling logic on a small set of static inputs before production. This enhancement facilitates test-driven development, improving developer efficiency and code reliability. More information on dbt unit tests feature can be found [here](https://docs.getdbt.com/docs/build/unit-tests).
```
Comment (mirnawong1) [inline website/docs/docs/core/connect-data-platform/teradata-setup.md]:
```suggestion
dbt-teradata 1.8.0 and above versions support unit tests, allowing validation of SQL modeling logic on a small set of static inputs before production. This enhancement facilitates test-driven development, improving developer efficiency and code reliability. More information on dbt unit tests feature can be found [here](/docs/build/unit-tests).
```

**Comment by tallamohan** [comment]
Hi @dataders  and @mirnawong1,
Can you please re-review this PR?

Thanks,
Mohan

File: website/docs/docs/core/connect-data-platform/teradata-setup.md [hunk 3]
```diff
@@ -124,6 +126,7 @@ Parameter               | Default     | Type           | Description
...
 `sslprotocol`           | `"TLSv1.2"` | string         | Specifies the TLS protocol for HTTPS/TLS connections. Equivalent to the Teradata JDBC Driver `SSLPROTOCOL` connection parameter.
 `teradata_values`       | `"true"`    | quoted boolean | Controls whether `str` or a more specific Python data type is used for certain result set column value types.
+`query_band`            | `"org=teradata-internal-telem;appname=dbt;"`    | string | Specifies the Query Band string to be set for each SQL request.
 
 For the full description of the connection parameters see https://github.com/Teradata/python-driver#connection-parameters.
```
Comment (mirnawong1) [inline website/docs/docs/core/connect-data-platform/teradata-setup.md]:
```suggestion
Refer to [connection parameters](https://github.com/Teradata/python-driver#connection-parameters) for the full description of the connection parameters.
```

File: website/docs/docs/core/connect-data-platform/teradata-setup.md [hunk 4]
```diff
@@ -124,8 +126,9 @@ Parameter               | Default     | Type           | Description
...
 `sslprotocol`           | `"TLSv1.2"` | string         | Specifies the TLS protocol for HTTPS/TLS connections. Equivalent to the Teradata JDBC Driver `SSLPROTOCOL` connection parameter.
 `teradata_values`       | `"true"`    | quoted boolean | Controls whether `str` or a more specific Python data type is used for certain result set column value types.
+`query_band`            | `"org=teradata-internal-telem;appname=dbt;"`    | string | Specifies the Query Band string to be set for each SQL request.
 
-For the full description of the connection parameters see https://github.com/Teradata/python-driver#connection-parameters.
+Refer to [connection parameters](https://github.com/Teradata/python-driver#connection-parameters) for the full description of the connection parameters.
 
 ## Supported Features
```
Comment (mirnawong1) [inline website/docs/docs/core/connect-data-platform/teradata-setup.md]:
```suggestion
## Supported features
```

File: website/docs/docs/core/connect-data-platform/teradata-setup.md [hunk 5]
```diff
@@ -141,8 +144,10 @@ The following incremental materialization strategies are supported:
 * `append` (default)
 * `delete+insert`
 * `merge`
+* `valid_history` (early access)
 
-To learn more about dbt incremental strategies please check [the dbt incremental strategy documentation](/docs/build/incremental-strategy).
+> * To learn more about dbt incremental strategies please check [the dbt incremental strategy documentation](/docs/build/incremental-strategy).
```
Comment (mirnawong1) [inline website/docs/docs/core/connect-data-platform/teradata-setup.md]:
```suggestion
:::info
- To learn more about dbt incremental strategies, refer to [the dbt incremental strategy documentation](/docs/build/incremental-strategy).
```

File: website/docs/docs/core/connect-data-platform/teradata-setup.md [hunk 6]
```diff
@@ -141,8 +144,11 @@ The following incremental materialization strategies are supported:
 * `append` (default)
 * `delete+insert`
 * `merge`
+* `valid_history` (early access)
 
-To learn more about dbt incremental strategies please check [the dbt incremental strategy documentation](/docs/build/incremental-strategy).
+:::info
+- To learn more about dbt incremental strategies, refer to [the dbt incremental strategy documentation](/docs/build/incremental-strategy).
+> * To learn more about `valid_history` incremental strategy please check [Teradata configs](/reference/resource-configs/teradata-configs).
```
Comment (mirnawong1) [inline website/docs/docs/core/connect-data-platform/teradata-setup.md]:
```suggestion
- To learn more about `valid_history` incremental strategy, refer to [Teradata configs](/reference/resource-configs/teradata-configs).
:::
```

File: website/docs/docs/core/connect-data-platform/teradata-setup.md [hunk 7]
```diff
@@ -225,6 +232,12 @@ For using cross-DB macros, teradata-utils as a macro namespace will not be used,
 
   `last_day` in `teradata_utils`, unlike the corresponding macro in `dbt_utils`, doesn't support `quarter` datepart.
 
+<VersionBlock firstVersion="1.8">
+
+dbt-teradata 1.8.0 and above versions support unit tests, allowing validation of SQL modeling logic on a small set of static inputs before production. This enhancement facilitates test-driven development, improving developer efficiency and code reliability. More information on dbt unit tests feature can be found [here](/docs/build/unit-tests).
```
Comment (mirnawong1) [inline website/docs/docs/core/connect-data-platform/teradata-setup.md]:
```suggestion
dbt-teradata 1.8.0 and later versions support unit tests, enabling you to validate SQL models and logic with a small set of static inputs before going to production. This feature enhances test-driven development and boosts developer efficiency and code reliability. Learn more about dbt unit tests [here](/docs/build/unit-tests).

```

File: website/docs/reference/resource-configs/teradata-configs.md [hunk 8]
```diff
@@ -262,6 +262,26 @@ Loading CSVs using dbt's seed functionality is not performant for large files. C
         +use_fastload: true
     ```
 
+## Snapshots
+Snapshots uses the HASHROW function of the Teradata database to generate a unique hash value for the 'dbt_scd_id' column. If you want to use your own hash UDF, there is a configuration option in the snapshot model called 'snapshot_hash_udf', which defaults to HASHROW. You can provide a value like <database_name.hash_udf_name>. If only hash_udf_name is provided, it uses the same schema as the model runs.
```
Comment (mirnawong1) [inline website/docs/reference/resource-configs/teradata-configs.md]:
```suggestion

Snapshots use the [HASHROW function](https://docs.teradata.com/r/Enterprise_IntelliFlex_VMware/SQL-Functions-Expressions-and-Predicates/Hash-Related-Functions/HASHROW/HASHROW-Function-Syntax) of the Teradata database to generate a unique hash value for the `dbt_scd_id` column. 

To use your own hash UDF, there is a configuration option in the snapshot model called `snapshot_hash_udf`, which defaults to HASHROW. You can provide a value like `<database_name.hash_udf_name>`. If you only provide `hash_udf_name`, it uses the same schema as the model runs.
```

File: website/docs/reference/resource-configs/teradata-configs.md [hunk 9]
```diff
@@ -262,6 +262,29 @@ Loading CSVs using dbt's seed functionality is not performant for large files. C
         +use_fastload: true
     ```
 
+## Snapshots
+
+Snapshots use the [HASHROW function](https://docs.teradata.com/r/Enterprise_IntelliFlex_VMware/SQL-Functions-Expressions-and-Predicates/Hash-Related-Functions/HASHROW/HASHROW-Function-Syntax) of the Teradata database to generate a unique hash value for the `dbt_scd_id` column. 
+
+To use your own hash UDF, there is a configuration option in the snapshot model called `snapshot_hash_udf`, which defaults to HASHROW. You can provide a value like `<database_name.hash_udf_name>`. If you only provide `hash_udf_name`, it uses the same schema as the model runs.
+
+for e.g. :
```
Comment (mirnawong1) [inline website/docs/reference/resource-configs/teradata-configs.md]:
```suggestion
For example, in the `snapshots/snapshot_example.sql` file:

```

File: website/docs/reference/resource-configs/teradata-configs.md [hunk 10]
```diff
@@ -262,6 +262,30 @@ Loading CSVs using dbt's seed functionality is not performant for large files. C
...
 
+## Snapshots
+
+Snapshots use the [HASHROW function](https://docs.teradata.com/r/Enterprise_IntelliFlex_VMware/SQL-Functions-Expressions-and-Predicates/Hash-Related-Functions/HASHROW/HASHROW-Function-Syntax) of the Teradata database to generate a unique hash value for the `dbt_scd_id` column. 
+
+To use your own hash UDF, there is a configuration option in the snapshot model called `snapshot_hash_udf`, which defaults to HASHROW. You can provide a value like `<database_name.hash_udf_name>`. If you only provide `hash_udf_name`, it uses the same schema as the model runs.
+
+For example, in the `snapshots/snapshot_example.sql` file:
+
+  snapshots/snapshot_example.sql
```
Comment (mirnawong1) [inline website/docs/reference/resource-configs/teradata-configs.md]:
```suggestion
```

File: website/docs/reference/resource-configs/teradata-configs.md [hunk 11]
```diff
@@ -291,6 +314,106 @@ Another e.g. for adding multiple grants:
 
 More on Grants can be found at https://docs.getdbt.com/reference/resource-configs/grants
```
Comment (mirnawong1) [inline website/docs/reference/resource-configs/teradata-configs.md]:
```suggestion
Refer to [grants](/reference/resource-configs/grants) for more information on Grants.
```

File: website/docs/reference/resource-configs/teradata-configs.md [hunk 12]
```diff
@@ -289,7 +312,112 @@ Another e.g. for adding multiple grants:
   ```
 > :information_source: `copy_grants` is not supported in Teradata.
 
-More on Grants can be found at https://docs.getdbt.com/reference/resource-configs/grants
+Refer to [grants](/reference/resource-configs/grants) for more information on Grants.
+
+## Query Band
+Query Band in dbt-teradata can be set on three levels:
```
Comment (mirnawong1) [inline website/docs/reference/resource-configs/teradata-configs.md:318]: adjusted this section as the formatting was off

Screenshot 2024-07-02 at 12 22 43

File: website/docs/reference/resource-configs/teradata-configs.md [hunk 13]
```diff
@@ -289,7 +312,112 @@ Another e.g. for adding multiple grants:
...
+  ```yaml
+  models:
+  Project_name:
+    +query_band: "app=dbt;model={model};"
+  ````
+
+- For example, if the model the user is running is `stg_orders`, `{model}` will be replaced with `stg_orders` in runtime.
+- If no `query_band` is set by the user, the default query_band used will be: ```org=teradata-internal-telem;appname=dbt;```
+
+## valid_history incremental materialization strategy (early access) {valid_history-incremental-materialization-strategy}
```
Comment (mirnawong1) [inline website/docs/reference/resource-configs/teradata-configs.md]:
```suggestion
## valid_history incremental materialization strategy (early access) {#valid-history-incremental-materialization-strategy}
```

**Review by mirnawong1 — APPROVED** [review]
this looks good to me from the editorial standpoint. cc @dataders what do you think?

File: website/docs/reference/resource-configs/teradata-configs.md [hunk 14]
```diff
@@ -289,7 +312,112 @@ Another e.g. for adding multiple grants:
...
+  ```yaml
+  models:
+  Project_name:
+    +query_band: "app=dbt;model={model};"
+  ````
+
+- For example, if the model the user is running is `stg_orders`, `{model}` will be replaced with `stg_orders` in runtime.
+- If no `query_band` is set by the user, the default query_band used will be: ```org=teradata-internal-telem;appname=dbt;```
+
+## valid_history incremental materialization strategy (early access) {#valid-history-incremental-materialization-strategy}
```
Comment (mirnawong1) [inline website/docs/reference/resource-configs/teradata-configs.md]: suggesting removing the 'early access' from the header

```suggestion
## valid_history incremental materialization strategy
_This is available in early access_
```

**Review by dataders — APPROVED** [review]
thanks @tallamohan! No reservations on my end -- ready to merge

**Comment by mirnawong1** [comment]
thanks so much @dataders ! and thanks for opening this pr up @tallamohan! let me know if yo have any thoughts or questions, looking forward to future contributions ✨!

**Pull request merged**